Fix 53

zkamvar opened the pull request:
I've modified `transform_params()` in the following ways:

1. assigned the subsetting statements to descriptive names
2. disallowed logical params from being quoted in the first place
3. protected fig.width and fig.height from being quoted
4. replaced the {glue} statement with the equivalent `shQuote(var, type = "cmd")`

reprex:

``` r
library(tinkr)
f <- tempfile()
writeLines("```{r ggplot, fig.width=4.2}\n# code here\n```", f)
yarn$new(f)$show()
#> ```{r ggplot, fig.width=4.2}
#> # code here
#> ```
```

<sup>Created on 2021-09-14 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

This will fix #53

maelle commented on R/utils.R:
```diff
@@ -61,33 +61,27 @@ transform_params <- function(params){
                        silent = TRUE)
 
-  if(inherits(params_string, "try-error")){
+  if (inherits(params_string, "try-error")){
```
```suggestion
  if (inherits(params_string, "try-error")) {
```

Even more whitespace :grin:

zkamvar replied:
good catch!

maelle commented on R/utils.R:
```diff
@@ -61,33 +61,27 @@ transform_params <- function(params){
-           !names(result) %in% c("language", "name")]}"')
 
-  logical_options <- c("echo", "eval", "collapse", "warning",
+  logical_options  <- c("echo", "eval", "collapse", "warning",
```
```suggestion
  logical_options <- c("echo", "eval", "collapse", "warning",
```

maelle reviewed (approved):
Thank you for catching and fixing this!

I only wonder whether we should add two options for letting users grow the list of unquoted and logical parameters (in case they are dealing with documents where they defined and use parameters)? Or [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it)*?

*: phrase I learnt last week. :sweat_smile:

zkamvar commented:
> I only wonder whether we should add two options for letting users grow the list of unquoted and logical parameters (in case they are dealing with documents where they defined and use parameters)? Or YAGNI*?

I'm actually thinking that we could simplify this a bit more. We only need to check for character arguments so that we can add an extra layer of quotes. Instead of explicitly creating a list of params to _not_ quote, we could map over the initial list (before unlisting) and find the characters that need quoting. I'll add another commit for this.

zkamvar commented:
Because I changed a part of the code that was not previously covered by tests and now that I changed it, codecov is complaining because this line is not being covered :weary: It would be good to get this line covered. Do you know what situation we would run into this? it seems to be a missing comma issue, but I'm unsure of how to trigger it.

maelle commented:
sorry which line is that exactly? I think I'm not looking at the right thing.

zkamvar commented:
> sorry which line is that exactly? I think I'm not looking at the right thing.

It's line 67 which is only run if the first pass through parsing params fails: https://github.com/ropensci/tinkr/pull/54/files#diff-60bef5af43a1bd541299ce9bceeb226afb23deaa5e637b145c85e05ffd552b0cR67

maelle commented:
mmmh, fine to ignore?

zkamvar commented:
> mmmh, fine to ignore?

I think so. In the future, it would be good to revisit this to figure out what situations would cause this block to be called.
